pkg/connector/group.go

```diff
@@ -175,7 +175,24 @@ func (o *groupResourceType) Grants(
 	case resourceTypeRole.Id:
 		roles, resp, err := listGroupAssignedRoles(ctx, o.connector.client, groupID, nil)
 		if err != nil {
-			return nil, "", nil, err
+			if resp != nil {
```

Comment:
I think it's a little easier to read if this logic is inverted. eg: if resp == nil, return the error on line 195. Then all the other code won't be indented as much.